nntrainer/layers/layer_context.cpp

```diff
@@ -308,4 +333,86 @@ bool RunLayerContext::readyToUse() const {
   return !inputs[0]->getVariable().empty();
 }
 
+/**
+ * @brief   validates the run context after run
+ *
+ * @return true if ready, else false
+ */
+bool RunLayerContext::validate(bool skip_input, bool skip_label) {
+  /**
+   * @note a common mistake when using run_context is re-assigning the tensor
+   * references which leads to nasty bugs. This validation ensures that the
+   * tensors are not set mistakenly by verifying their unique names
+   */
+#ifdef ENABLE_TEST
+  if (tensor_map.empty() || !tensor_map[inputs[0]->getName()]) {
+    auto filler = [this](const Var_Grad *val) {
+      tensor_map[val->getName()] = val->getVariableRef().getData();
+      tensor_map[val->getGradientName()] = val->getGradientRef().getData();
+    };
+
+    auto filler_w = [this, filler](const std::vector<Weight *> &vec) {
```

Comment:
Code clean idea:

```suggestion
    auto filler_w = [this, filler](const auto &vec) { 
```

and removing filler_vw